src/components/Navigation/Navigation.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+import ProgressBar from './progressBar';
+import NextArrow from './NextArrow';
+
+function Navigation(props) {
+  const { children, progressBarColor, transitionTime } = props;
+
+  const [pageIndex, changePageIndex] = useState(0);
+  const [isScrolling, changeScrollState] = useState(false);
+
+  const maxPages = Object.keys(children).length - 1;
+
+  const pageContainer = useRef(null);
+
+  useEffect(() => {
+    pageContainer.current.style.transform = `translate(0, ${pageIndex * -100}%)`;
+  });
+
+  const pageUp = () => {
+    if (pageIndex > 0) {
+      changePageIndex(pageIndex - 1);
+    }
+  };
+
+  const pageDown = () => {
+    if (pageIndex < maxPages) {
+      changePageIndex(pageIndex + 1);
+    }
+  };
+
+  const onScroll = (event) => {
+    if (!isScrolling) {
+      changeScrollState(true);
+
+      if (event.deltaY < 0) {
+        pageUp();
+      } else {
+        pageDown();
+      }
+
+      setTimeout(() => {
+        changeScrollState(false);
+      }, transitionTime);
+    }
+  };
+
+  /*
+  const handleKeyDown = (event) => {
+    if (event.keyCode === 38) {
+      pageUp()
+    }
+    if (event.keyCode === 40) {
+      pageDown()
+    }
+  }
```

Comment:
If possible, try to not submit commented out code to PRs. Obviously if there's a good reason for it then feel free.